### ISSUE TITLE
Create Speaker Button

### ISSUE THREAD

**Issue by tomjn** [imath/wordcamp-talks#41]
When on a proposal, if a speaker post for the author does not exist, a button should allow the creation of a draft speaker post

**Comment by imath** [comment]
This means we need to remove the user's profile from the plugin ? What is the benefit of a post type for a user, it seems really weird to me to consider a user as content. But if there's a consensus about this post type, then i surrender.

**Comment by tomjn** [comment]
My understanding is that wordcamp.org already has such a post type. If it
isn't present we can link to the author page

**Comment by imath** [comment]
Tbh, although i have this difficulty to imagine managing users like content, I'm fine with the post type as it's part of WordCamp sites' package for a long time.

FYI, the plugin is adding profile pages for users on the front-end. It has some tabs to list profile informations, the talks he submitted, rated, commented, or list the talks the rater still have to review.

So here's another opportunity to remove lines of code! It should make it lighter and will allow us to stop using the template loading stuff i was using in favour of the one provided by default by WordPress for post types.

**Comment by iandunn** [comment]
> My understanding is that wordcamp.org already has such a post type.

That's correct.

>  i have this difficulty to imagine managing users like content

They're not actually users. They don't have a role on the site and can't log in to wp-admin. If they have a WordPress.org account, then they do have an account on the network, but not a role for the individual camp site (unless they also happen to be an organizer).

**Comment by iandunn** [comment]
Should this be a separate issue from #40? I don't see a use case for wanting to approve a session without its corresponding speaker, or vice-versa. Wouldn't we want to create them both at the same time, once the admin has decided to accept the session?

**Comment by jennybeaumont** [comment]
I agree that it makes sense to have these two actions combined into one. I also see the value of retaining speaker CPT whether or not applicants connect via their .org account. As @iandunn points out, they are not specifically site users - we've already discussed bypassing this and their admin access with front-end connection/signup.

**Comment by tomjn** [comment]
This only makes sense in a WordCamp site context, and the embedded user profiles have already been removed from this plugin. No additional post types etc are being added, the data model remains the same.

All the button does is automate creating a speaker post, something WordCamp Organisers already do manually

**Comment by tomjn** [comment]
As for #40, no, someone might be speaking more than once, or a speaker post might already exist, in which case we'd need to re-do the wording of the button, and add conditional logic etc, I would err on the side of flexibility, and if it turns out they should be merged, then that's an easier thing to do once this and #40 are merged. It also simplifies testing
